exercises/00-init/src/modules/root/components/users-list.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
+class UsersList extends React.Component {
+    constructor() {
+        super();
+        this.state = { 
+            users : []
+        }
+    }
+
+    render() {
+        return <div>
+            <button onClick={ () => this.addUser('Arya', 'Stark')}>Add Arya</button>
+            <button onClick={ () => this.addUser('Daenerys', 'Targaryen')}>Add Daenerys</button>
+            <br/>
+            {this.state.users.length === 0 ? 
+                "No users" :
+                this.state.users.map( ({id, firstName, lastName}) => 
+                    <li key={id}>{firstName} {lastName}</li>)
+            }
+        </div>
+    }
+
+    addUser(firstName, lastName) {
+        this.setState( 
+            {users: this.state.users.concat({id: this.state.users.length, firstName: firstName, lastName: lastName}) }

```

Comment:
Although this works without any issue, it is more common to use the spread operator for this, like so:
```
this.setState({
  users: [
    ...this.state.users,
    { id: this.state.users.length, firstName, lastName }
  ]
})
```
The added indentation is just for readability, not necessary.
Also, I forgot to mention it, but notice in my suggestion how it is possible to shorten the assignment of `firstName` and `lastName` - if the key is the same string as is the name of the assigned variable, you can write it like that.

Reply:
I'm not very familiar with the spread operator, so it didn't even occur to me to use it. I'll rewrite it like that and keep it in mind.